packages/modal/src/ux-modal.ts

```diff
@@ -0,0 +1,361 @@
+import { UxModalService, UxModalServiceResult } from './ux-modal-service';
+import { customElement, bindable } from 'aurelia-templating';
+import { inject } from 'aurelia-dependency-injection';
+import { StyleEngine, UxComponent } from '@aurelia-ux/core';
+import { UxModalTheme } from './ux-modal-theme';
+import { TaskQueue } from 'aurelia-framework';
+import { PLATFORM, DOM } from 'aurelia-pal';
+import { getLogger } from 'aurelia-logging';
+import { UxModalPosition, UxModalKeybord, UxDefaultModalConfiguration } from './ux-modal-configuration';
+
+const log = getLogger('ux-modal');
+
+@inject(Element, StyleEngine, UxModalService, TaskQueue, UxDefaultModalConfiguration)
+@customElement('ux-modal')
+export class UxModal implements UxComponent {
+
+  @bindable public position: UxModalPosition = 'center';
+  @bindable public host: 'body' | HTMLElement | false | string = 'body';
+  @bindable public modalBreakpoint: number = 768;
```

Comment:
Probably minor. But we will need some accomodating comments for what things are later. It may not be as obvious as it seems, to anyone beside the author.